src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -417,6 +417,12 @@ private Task WriteFormattedInt32Async(int value, CancellationToken cancellationT
                     responseStream = EmptyReadStream.Instance;
                     ReturnConnectionToPool();
                 }
+                else if (request.Method == s_httpConnectMethod && (int)response.StatusCode == 200)
+                {
+                    // Successful response to CONNECT does not have body.
+                    // What ever comes next should be opaque.
+                    responseStream = EmptyReadStream.Instance;
```

Comment:
I originally did that @geoffkizer but await responseTask = client.SendAsync(request) blocked forever as it tried to read body. When I was thinking about it, the rest of the stream is really not part of the response to CONNECT. I can explore that more.

Reply:
Yeah, that's because HttpClient buffers responses by default.  For now at least, you can avoid this by using HttpCompletionOption.ResponseHeadersRead.

That said, it doesn't seem optimal to hang any CONNECT request if you don't use ResponseHeadersRead.  This should probably be implicit.